src/Pact/Gas.hs

```diff
@@ -18,85 +18,96 @@ module Pact.Gas
 import Control.Monad.State.Strict
 import Data.Text
 import Data.IORef
+import Pact.Types.Gas
 import Pact.Types.Runtime
 import Control.Lens
 import Control.Arrow
 import Data.Word
 import Pact.Types.Pretty
 
 -- | Compute gas for some application or evaluation.
-computeGas :: Either (Info,Text) FunApp -> GasArgs -> Eval e Gas
+computeGas :: Either (Info,Text) FunApp -> GasArgs -> Eval e ()
 computeGas i args = do
   GasEnv {..} <- view eeGasEnv
   g0 <- getGas
   let
     (info,name) = either id (_faInfo &&& _faName) i
     g1 = runGasModel _geGasModel name args
-  let gUsed = g0 + g1
-  evalLogGas %= fmap ((msg name gUsed, g1):)
-  putGas gUsed
-  if gUsed > fromIntegral _geGasLimit then
-    throwErr GasError info $ "Gas limit (" <> pretty _geGasLimit <> ") exceeded: " <> pretty gUsed
-    else return gUsed
- where
-   msg name used = renderCompactText' (pretty name <> ":" <> pretty args <> ":currTotalGas=" <> pretty used)
+  let gMillisUsed = g0 <> g1
+      gUsed = milliGasToGas gMillisUsed
+  evalLogGas %= fmap ((msg name gUsed, milliGasToGas g1):)
```

Comment:
Isn't this the same as:
```
evalLogGas.mapped %= ((msg name gUsed, milliGasToGas g1):)
```
?

Reply:
Adding more lens golf: `views (foo.bar) $ Set.member ..` can be condensed to `views (foo.bar.at) ...` or `ix` if we're pattern matching.